src/util.ts

```diff
@@ -21,14 +25,26 @@ export const AuthorityPrefix = "coder-vscode"
  * Throw an error if the host is invalid.
  */
 export function parseRemoteAuthority(authority: string): AuthorityParts | null {
-  // The authority looks like: vscode://ssh-remote+<ssh host name>
-  const authorityParts = authority.split("+")
+  // The Dev Container authority looks like: vscode://attached-container+containerNameHex@ssh-remote+<ssh host name>
+  // The SSH authority looks like: vscode://ssh-remote+<ssh host name>
+  const authorityParts = authority.split("@")
```

Comment:
I wonder if `new URL(authority)` would be more robust here for parsing?  The `attached-container+containerNameHex` part would be in `url.username`, and `ssh-remote+<ssh host name>` would be in `url.host`.